resources/assets/js/features/user/actions.js

```diff
@@ -0,0 +1,82 @@
+import * as actions from './actionTypes';
+import axios from 'axios';
+
+export const registerSuccess = data => ({
+    type: actions.USER_REGISTER_SUCCESS,
+    data
+});
+
+export const registerFailed = data => ({
+    type: actions.USER_REGISTER_FAILED,
+    data
+});
+
+export const doRegister = (data) => {
+    return dispatch => {
+        axios.post('/api/user/register', data)
+            .then(response => dispatch(registerSuccess(response.data)))
+            .catch(error => dispatch(registerFailed(error.response.data)));
+    };
+};
+
+export const registerValidate = data => {
```

Comment:
move validator to separate service like UserValidator or UserService or something like that